### PR TITLE
fix(react) - Refactor cardgroup to use individual component

### PR DESCRIPTION
**Pull request by Shashika6**
Fixes https://github.com/international-labour-organization/designsystem/issues/869

### Notes :- 

The cardgroup component design is broken and it needs to use the individual card components instead of the common card component

### Fix :- 

* Sync classnames with twig
* Use individual card components

### Screenshots :- 

Screenshot 2024-03-14 at 13 42 19

### CODE REVIEW

**Comment by justintemps** [comment]
Nice work @Shashika6

File: packages/react/src/components/Cards/CardGroup.props.ts [hunk 1]
```diff
@@ -1,12 +1,31 @@
...
+  | StatCardProps
+  | TextCardProps;
 
 export interface CardGroupProps {
   /**
    * An array of card objects
    */
-  cards?: Required<CardProps>[];
-
+  cards: CardProps[];
```
Comment (bashlk) [inline packages/react/src/components/Cards/CardGroup.props.ts:28]: This type declaration will also allow different card types to be used at once. The CardGroup is only supposed to be used with a single type of card at once right?

File: packages/react/src/components/Cards/CardGroup.tsx [hunk 2]
```diff
@@ -2,40 +2,53 @@ import { FC } from "react";
...
+  DataCard,
+  DetailCard,
+  FactlistCard,
+  MultilinkCard,
+  PromoCard,
+  StatCard,
+  TextCard,
+} from ".";
+
+const cardMapper: Record<string, FC<any>> = {
```
Comment (bashlk) [inline packages/react/src/components/Cards/CardGroup.tsx:27]: Why FC<any> and not simply FC?

File: packages/react/src/stories/CardGroup/CardGroup.mdx [hunk 3]
```diff
@@ -9,7 +9,7 @@ import { CardGroup } from "../../components";
 Card Group displays a group of Cards.
 
 <Canvas>
-  <Story of={CardGroupStories.GraphicCardGroup} />
+  <Story of={CardGroupStories.FeatureCardGroup} />
```
Comment (bashlk) [inline packages/react/src/stories/CardGroup/CardGroup.mdx:12]: Why change the default story?

**Review by bashlk — CHANGES_REQUESTED** [review]

File: packages/react/src/components/Cards/CardGroup.props.ts (hunk 1 above)
Comment (Shashika6) [reply packages/react/src/components/Cards/CardGroup.props.ts:28]: Yes @bashlk, I added a union type here to make sure that the mapper card component will have one of the above types any suggestions ?

File: packages/react/src/stories/CardGroup/CardGroup.mdx (hunk 3 above)
Comment (Shashika6) [reply packages/react/src/stories/CardGroup/CardGroup.mdx:12]: Changed it to be in the same order as twig, in the twig its defaulting to feature card.

File: packages/react/src/components/Cards/CardGroup.tsx (hunk 2 above)
Comment (Shashika6) [reply packages/react/src/components/Cards/CardGroup.tsx:27]: Hmm didn't get you here @bashlk

File: packages/react/src/components/Cards/Card.props.ts
[the remark this answers is not in the document]
Comment (Shashika6) [reply packages/react/src/components/Cards/Card.props.ts:1]: Will do @bashlk

File: packages/react/src/components/Cards/CardGroup.tsx (hunk 2 above)
Comment (bashlk) [reply packages/react/src/components/Cards/CardGroup.tsx]: Uff Github ate my angle brackets 😅 - should make more sense now.

File: packages/react/src/components/Cards/CardGroup.props.ts (hunk 1 above)
Comment (bashlk) [reply packages/react/src/components/Cards/CardGroup.props.ts]: I mean something like this

```
export interface CommonCardGroupProps {
  cardCount: Required<string>;
  cta?: LinkProps;
}

export interface StatCardGroup extends CommonCardGroupProps {
  type: "stat";
  cards: StatCardProps[];
}

export interface MultilinkCardGroup extends CommonCardGroupProps {
  type: "multilink";
  cards: MultilinkCardProps[];
}

...

export type CardGroupProps =
  | StatCardGroup
  | MultilinkCardGroup
  | TextCardGroup
  | PromoCardGroup
  | FeatureCardGroup
  | DetailCardGroup
  | FactlistCardGroup
  | DataCardGroup;
```

With this, someone won't be able to provide cards of a different type than the one specified as the type to the CardGroup.

File: packages/react/src/components/Cards/CardGroup.tsx (hunk 2 above)
Comment (bashlk) [reply packages/react/src/components/Cards/CardGroup.tsx]: Also on a second look cardMapper doesn't need to declare components / functions - you can have it return the components directly

e.g.
```
const cardMapper = {
     stat: StatCard,
     multilink: MultiLinkCard,
     ...
}
```

File: packages/react/src/components/Cards/CardGroup.props.ts (hunk 1 above)
Comment (bashlk) [reply packages/react/src/components/Cards/CardGroup.props.ts]: Also note that card count right now is not camel case. It probably should be. 

Also I don't think it is required at all. We can count the number of items passed into the cards prop to get the number of cards.

Comment (Shashika6) [reply packages/react/src/components/Cards/CardGroup.props.ts]: The cardCount here refers to the amount of cards per row @bashlk. I think it makes sense to refactor the name of that variable right ?

**Comment by Shashika6** [comment]
@justintemps This is ready now, @bashlk made the changes you requested as well

**Comment by justintemps** [comment]
Hey @Shashika6, I think you still need to add the doc block quotes so they'll appear in the `prop` descriptions. You do this literally just by adding comments to the props in the `props.ts` files. See `Card.props.ts` for an example.

**Comment by Shashika6** [comment]
Hey @justintemps added the prop descriptions as well.

**Pull request merged**